lib/core/include/irods/transport/default_transport.hpp

```diff
@@ -1,5 +1,16 @@
-#ifndef IRODS_IO_DEFAULT_TRANSPORT_HPP
-#define IRODS_IO_DEFAULT_TRANSPORT_HPP
+#undef IRODS_IO_DEFAULT_TRANSPORT_HPP_INCLUDE_HEADER
```

Comment:
Are `_INCLUDE_HEADER` suffixes meant to be how we want to do things moving forward, or is this a special case?

Reply:
This PR is a special case.

Because it uses macros such as `rxComm`, I had to figure out a way to allow the same headers to be included multiple times while maintaining the same behavior as the include guards.

The preprocessor logic mirrors the behavior of normal include guards while also allowing the contents of the headers to change based on whether the special `_ENABLE_SERVER_API` macro is defined or not.